File: frontend/src/api/portfolio.ts
import api from './index'

export interface Portfolio {
  id: string
  name: string
  initial_capital: number
  cash: number
  position: number
  buy_fee_rate: number
  sell_fee_rate: number
  stamp_tax_rate: number
  min_fee: number
}

export const portfolioApi = {
  list: () => api.get<Portfolio[]>('/portfolios'),
  get: (id: string) => api.get<Portfolio>(`/portfolios/${id}`),
  create: (data: Partial<Portfolio>) => api.post<Portfolio>('/portfolios', data),
  update: (id: string, data: Partial<Portfolio>) => api.put(`/portfolios/${id}`, data),
  delete: (id: string) => api.delete(`/portfolios/${id}`),
}

File: backend/src/trade_alpha/dao/portfolio.py
"""AccountConfig Document model."""

from datetime import datetime
from typing import Optional
from pydantic import Field
from beanie import Document, PydanticObjectId


class AccountConfig(Document):
    """Account config document for MongoDB."""

    name: str
    initial_capital: float
    buy_fee_rate: float = Field(default=0.0003)
    sell_fee_rate: float = Field(default=0.0003)
    stamp_tax_rate: float = Field(default=0.001)
    min_fee: float = Field(default=5.0)
    cash: float = Field(default=0.0)
    position: int = Field(default=0)
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Settings:
        collection = "account_configs"
        indexes = [
            "name",
        ]

    def __init__(self, **data):
        super().__init__(**data)
        if self.cash == 0.0:
            self.cash = self.initial_capital
